haas/errors.py

```diff
@@ -92,7 +94,10 @@ class BlockedError(APIError):
     some other change.  For example, deletion is blocked until the components
     are deleted, and possibly until the dirty flag is cleared as well.
     """
-    status_code = 409  # Conflict
+    status_code = 423  # was 409,
```

Comment:
Similarly, this is part of WEBDAV, and has some specific constraints on its use:

> The 423 (Locked) status code means the source or destination resource of a method is locked. **This response SHOULD contain an appropriate precondition or postcondition code, such as 'lock-token-submitted' or 'no-conflicting-lock'.**

(Again, emphasis mine).

The way we use BlockedError isn't really the same as a resource being "locked." 409 Conflict is still the best thing I've seen; you can only get so descriptive with HTTP status codes.